tests/unit/evaluation/CrossValidatedCalibration_unittest.cc

```diff
@@ -0,0 +1,65 @@
+#include "environments/MultiLabelTestEnvironment.h"
+#include <gtest/gtest.h>
+#include <shogun/distance/EuclideanDistance.h>
+#include <shogun/evaluation/CrossValidatedCalibration.h>
+#include <shogun/evaluation/MulticlassAccuracy.h>
+#include <shogun/evaluation/MulticlassAccuracy.h>
+#include <shogun/evaluation/SigmoidCalibrationMethod.h>
+#include <shogun/evaluation/StratifiedCrossValidationSplitting.h>
+#include <shogun/features/DataGenerator.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/MulticlassLabels.h>
+#include <shogun/multiclass/MulticlassOCAS.h>
+
+using namespace shogun;
+
+extern MultiLabelTestEnvironment* multilabel_test_env;
+
+#ifdef HAVE_LAPACK
+TEST(CrossValidatedCalibrationTest, check_probability_sum)
+{
+	CMath::init_random(5);
+	float64_t C = 1.0;
+	std::shared_ptr<GaussianCheckerboard> mockData =
+	    multilabel_test_env->getMulticlassFixture();
+
+	CDenseFeatures<float64_t>* train_feats = mockData->get_features_train();
+	CDenseFeatures<float64_t>* test_feats = mockData->get_features_test();
+	CMulticlassLabels* ground_truth =
+	    (CMulticlassLabels*)mockData->get_labels_test();
+	index_t n_folds = 4;
+	CStratifiedCrossValidationSplitting* splitting =
+	    new CStratifiedCrossValidationSplitting(ground_truth, n_folds);
+	CMulticlassOCAS* mocas = new CMulticlassOCAS(C, train_feats, ground_truth);
+	mocas->set_epsilon(1e-5);
+	mocas->parallel->set_num_threads(1);
+
+	CCrossValidatedCalibration* cross = new CCrossValidatedCalibration(
+	    mocas, ground_truth, splitting, new CSigmoidCalibrationMethod);
+
+	cross->train(train_feats);
+
+	CMulticlassLabels* pred = (CMulticlassLabels*)cross->apply(test_feats);
+
+	index_t num_classes = pred->get_num_classes();
+
+	SGVector<float64_t> confidence_sums;
+	for (index_t i = 0; i < num_classes; ++i)
+	{
+		SGVector<float64_t> scores = pred->get_multiclass_confidences(i);
+		if (i == 0)
+			confidence_sums = scores;
```

Comment:
rather than writing unit tests here, a good sanity check would be to reproduce the plots in http://scikit-learn.org/stable/_images/sphx_glr_plot_calibration_curve_002.png
You could do that via say Python bindings. Once those work, then we can start adding more unit tests ...
Let me know what you think

Reply:
That looks like a great idea. I will try doing that.

Reply:
The python interfaces for these files are generated automatically right? Or do I have to add these files to some list? I am asking because I am not able to access these (CrossValidatedCalibration, Calibration, etc) from the shogun module in Python.

Reply:
@durovo yeah as of this state the python interface will not contain these methods.. you'll need to add it to some definition files
but that should be another PR. for the time being you should write unit tests for this and test it from the C++ API

Reply:
Okay